aws-sdk-resources/lib/aws-sdk-resources/services/s3/file_downloader.rb

```diff
@@ -104,9 +104,7 @@ def concatenate_files(fileparts)
         begin
           concatenate_parts(fileparts)
         ensure
-          fileparts.each do |part|
-            File.unlink(part) if File.exists?(part)
-          end
+          clean_up_parts(fileparts)
```

Comment:
Not needed here if this is run within the outer begin/ensure block which handles cleanup.